code_samples/api/rest_api/src/Rest/Serializer/GreetingInputDenormalizer.php

```diff
@@ -0,0 +1,41 @@
+<?php declare(strict_types=1);
+
+namespace App\Rest\Serializer;
+
+use App\Rest\Values\Greeting;
+use Symfony\Component\Serializer\Normalizer\DenormalizerAwareInterface;
+use Symfony\Component\Serializer\Normalizer\DenormalizerAwareTrait;
+use Symfony\Component\Serializer\Normalizer\DenormalizerInterface;
+
+class GreetingInputDenormalizer implements DenormalizerInterface, DenormalizerAwareInterface
+{
+    use DenormalizerAwareTrait;
+
+    public function denormalize(mixed $data, string $type, ?string $format = null, array $context = [])
+    {
+        if ('json' === $format) {
+            $data = $data[array_key_first($data)];
+        }
```

Comment:
Unwrap JSON.
I could even test that `'GreetingInput' === array_key_first($data)`